### PR TITLE
test_: run functional tests on host (no container)

### PR DESCRIPTION
**Pull request by antdanchenko**
- tests are running on host in CI instead of docker container
- we control docker from tests, every new StatusBackend instance == new container
- statusd is no more

### CODE REVIEW

File: tests-functional/clients/status_backend.py [hunk 1]
```diff
@@ -16,25 +17,59 @@
...
+                "--address", "0.0.0.0:3333",
+            ],
+            ports={"3333/tcp": host_port}
+        )
+
+        network = self.docker_client.networks.get(f"{docker_project_name}_default")
+        network.connect(container)
+
+        option.status_backend_containers.append(container.id)
+        time.sleep(2)
```
Comment (antdanchenko) [inline tests-functional/clients/status_backend.py]: TBD

**Comment by antdanchenko** [comment]
please note, codecov is generating report from a single container, I would appreciate to fix it asap in separate PR in order not to hold this one any longer

**Review by yevh-berdnyk — APPROVED** [review]

File: .gitignore [hunk 2]
```diff
@@ -126,6 +126,7 @@ internal/version/GIT_COMMIT
 internal/sentry/SENTRY_CONTEXT_NAME
 internal/sentry/SENTRY_CONTEXT_VERSION
 internal/sentry/SENTRY_PRODUCTION
+logs/status-backend/*
```
Comment (igor-sirotin) [inline .gitignore]: I don't see this path is used somewhere, do we really need to gitignore it?

File: tests-functional/clients/status_backend.py [hunk 3]
```diff
@@ -4,23 +4,32 @@
...
 
 class StatusBackend(RpcClient, SignalClient):
 
-    def __init__(self, await_signals=[], url=None):
-        try:
-            url = url if url else random.choice(option.status_backend_urls)
-        except IndexError:
-            raise Exception("Not enough status-backend containers, please add more")
-        option.status_backend_urls.remove(url)
+    def __init__(self, await_signals=[], new_container=True):
```
Comment (igor-sirotin) [inline tests-functional/clients/status_backend.py]: `new_container` doesn't seem to be used anywhere.

I guess this is the way devs can run tests with locally-running status-backend? 
Perhaps, we should do smth like this then?
```python
    if option.status_backend_url:
        url  = option.status_backend_url
    else:
        self.docker_client = docker.from_env()
        host_port = random.choice(option.status_backend_port_range)

        self.container = self._start_container(host_port)
        url = f"http://0.0.0.0:{host_port}"
        option.status_backend_port_range.remove(host_port)
```

And no need for the argument then

File: tests-functional/clients/status_backend.py [hunk 4]
```diff
@@ -4,23 +4,32 @@
...
+
+        if new_container:
+
+            self.docker_client = docker.from_env()
+            host_port = random.choice(option.status_backend_port_range)
+
+            self.container = self._start_container(host_port)
+            url = f"http://0.0.0.0:{host_port}"
```
Comment (igor-sirotin) [inline tests-functional/clients/status_backend.py]: Should be localhost:
```suggestion
            url = f"http://127.0.0.1:{host_port}"
```

File: tests-functional/clients/status_backend.py [hunk 5]
```diff
@@ -29,14 +38,71 @@ def __init__(self, await_signals=[], url=None):
...
+        network = self.docker_client.networks.get(
+            f"{docker_project_name}_default")
+        network.connect(container)
+
+        option.status_backend_containers.append(container.id)
+        return container
+
+    def _health_check(self):
+        start_time = time.time()
+        while True:
```
Comment (igor-sirotin) [inline tests-functional/clients/status_backend.py:90]: When do we quit this loop?
Maybe we could use `signals` websocket connection as health check? If disconnected with error -> raise exception.

File: tests-functional/clients/status_backend.py [hunk 6]
```diff
@@ -29,14 +38,71 @@ def __init__(self, await_signals=[], url=None):
...
+            "environment": {
+                "GOCOVERDIR": "/coverage/binary",
+            },
+            "volumes": {
+                coverage_path: {
+                    "bind": "/coverage/binary",
+                    "mode": "rw",
+                }
+            },
+            "stop_signal": "SIGINT"
```
Comment (igor-sirotin) [inline tests-functional/clients/status_backend.py]: Not needed anymore

File: tests-functional/conftest.py [hunk 7]
```diff
@@ -19,14 +19,10 @@ def pytest_addoption(parser):
...
-        "--status_backend_urls",
+        "--status_backend_url",
         action="store",
         help="",
-        default=[
-            f"http://0.0.0.0:{3314 + i}" for i in range(
-                int(os.getenv("STATUS_BACKEND_COUNT", 10))
-            )
-        ],
+        default="http://0.0.0.0:3334",
```
Comment (igor-sirotin) [inline tests-functional/conftest.py]: I think we should leave the default value empty now

File: tests-functional/conftest.py [hunk 8]
```diff
@@ -52,6 +60,24 @@ class Option:
...
+
     option.base_dir = os.path.dirname(os.path.abspath(__file__))
+
+def pytest_unconfigure():
+    docker_client = docker.from_env()
+    for container_id in option.status_backend_containers:
+        try:
+            container = docker_client.containers.get(container_id)
+            container.stop(timeout=30)
+            container.remove()
```
Comment (igor-sirotin) [inline tests-functional/conftest.py:75]: If I understand correctly, `pytest_unconfigure` is called when ALL tests have finished.
But maybe we can call `container.stop` at each test end? We could then free the used port as well.

File: tests-functional/tests/test_wallet_activity_session.py [hunk 9]
```diff
@@ -20,32 +17,34 @@ def validate_entry(entry, tx_data):
 
 @pytest.mark.wallet
 @pytest.mark.rpc
-class TestWalletActivitySession(SignalTestCase):
-    await_signals = ["wallet",
+@pytest.mark.takoe
```
Comment (igor-sirotin) [inline tests-functional/tests/test_wallet_activity_session.py]: leftover?

File: tests-functional/clients/status_backend.py (hunk 5 above)
Comment (antdanchenko) [reply tests-functional/clients/status_backend.py:90]: if we try to connect to the websockets right after the start it fails with `urllib3.exceptions.ProtocolError: ('Connection aborted.', ConnectionResetError(54, 'Connection reset by peer'))` and it is not very convenient to use it as health check cause of  [threading.Thread](https://github.com/status-im/status-go/blob/880d954657a0d51cb8de9864899ce4ab61cabd81/tests-functional/clients/status_backend.py#L43)

for docker-compose we had similar check https://github.com/status-im/status-go/blob/880d954657a0d51cb8de9864899ce4ab61cabd81/tests-functional/docker-compose.test.status-go.yml#L44

File: tests-functional/conftest.py (hunk 7 above)
Comment (antdanchenko) [reply tests-functional/conftest.py]: agree, and reverse condition for url so we can run status-backend with binary

File: tests-functional/conftest.py (hunk 8 above)
Comment (antdanchenko) [reply tests-functional/conftest.py:75]: we have many test case that are using single instance of status backend for many tests, we do not have a test == fresh container, yet

Comment (antdanchenko) [reply tests-functional/conftest.py:75]: as result to stop at each test end we need to start at each test start :) possible but I would rather do it outside this PR and review if we really need that many instances

File: tests-functional/clients/status_backend.py (hunk 5 above)
Comment (antdanchenko) [reply tests-functional/clients/status_backend.py:90]: we stop it in 20 seconds if there is an exception instead of a valid response

Comment (igor-sirotin) [reply tests-functional/clients/status_backend.py:90]: Alright, that's fair 👍 
And I will implement a `/health` endpoint for clarity

File: tests-functional/conftest.py (hunk 8 above)
Comment (igor-sirotin) [reply tests-functional/conftest.py:75]: Sure, let's do it in a separate PR 👍

File: tests-functional/clients/status_backend.py (hunk 5 above)
Comment (igor-sirotin) [reply tests-functional/clients/status_backend.py:90]: @antdanchenko here it is: 
- https://github.com/status-im/status-go/pull/6201

Let me know if it's useful. You don't have to use it in this PR though of course.

**Review by igor-sirotin — APPROVED** [review]
Good job 👏

**Pull request merged**